parts/kubernetesagentresourcesvmas.t

```diff
@@ -15,20 +15,26 @@
       "name": "[concat(variables('{{.Name}}VMNamePrefix'), 'nic-', copyIndex(variables('{{.Name}}Offset')))]",
       "properties": {
 {{if .IsCustomVNET}}
-	    "networkSecurityGroup": {
-		    "id": "[variables('nsgID')]"
-	    },
+        "networkSecurityGroup": {
+          "id": "[variables('nsgID')]"
+        },
 {{end}}
         "ipConfigurations": [
+          {{range $seq := loop 1 .IPAddressCount}}
           {
-            "name": "ipconfig1",
+            "name": "ipconfig{{$seq}}",
             "properties": {
+              {{if eq $seq 1}}
+              "primary": true,
+              {{end}}
               "privateIPAllocationMethod": "Dynamic",
               "subnet": {
-                "id": "[variables('{{.Name}}VnetSubnetID')]"
-             }
+                "id": "[variables('{{$.Name}}VnetSubnetID')]"
```

Comment:
Yes. Inside the new "range" block, ".Name" now refers to the Name field of the inner iterator value (so $seq.Name) instead of AgentPool (AgentPool.Name). So the "$" is required to refer to AgentPool.Name by reaching it through the global variable "$".